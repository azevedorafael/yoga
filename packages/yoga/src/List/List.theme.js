const List = ({ spacing, borders, colors }) => ({
  listItem: {
    padding: {
      top: spacing.large,
      right: spacing.xlarge,
      bottom: spacing.large,
      left: spacing.xlarge,
    },
  },
  border: {
    width: borders.small,
    color: colors.gray[3],
  },
});

export default List;
