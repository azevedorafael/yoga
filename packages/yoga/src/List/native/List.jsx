import React from 'react';
import styled from 'styled-components';
import { FlatList } from 'react-native';

const StyledFlatList = styled.FlatList.attrs(() => ({
  as: FlatList,
}))`
  width: 100%;
`;

/** Lists are a continuous group of text or images. They are composed of items containing primary and supplemental actions, which are represented by icons and text. */
const List = ({ ...rest }) => <StyledFlatList {...rest} />;

List.displayName = 'List';

export default List;
