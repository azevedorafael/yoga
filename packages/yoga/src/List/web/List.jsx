import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledList = styled.ul`
  ${({
    divided,
    horizontal,
    theme: {
      components: {
        list: {
          border: { width: borderWidth, color: borderColor },
        },
      },
    },
  }) => `
    display: flex;
    flex-direction: ${horizontal ? 'row' : 'column'};

    > * {
        ${
          divided
            ? `border-${
                horizontal ? 'right' : 'bottom'
              }: ${borderWidth}px solid ${borderColor}`
            : ''
        };

        &:last-child {
          border: none;
        }
      }
    }
  `}
`;

const List = ({ horizontal, divided, ...rest }) => (
  <StyledList horizontal={horizontal} divided={divided} {...rest} />
);

List.propTypes = {
  horizontal: PropTypes.bool,
  divided: PropTypes.bool,
};

List.defaultProps = {
  horizontal: false,
  divided: true,
};

List.displayName = 'List';

export default List;
