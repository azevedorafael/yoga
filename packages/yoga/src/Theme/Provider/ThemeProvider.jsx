import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import * as tokens from '@gympass/yoga-tokens';
import * as themes from '../themes';

const getTheme = ({ theme, locale }) => {
  const token = tokens[locale] || tokens.default;
  const appTheme = themes[theme] || themes.default;

  return appTheme(token);
};

const ThemeProvider = ({ children, ...theme }) => (
  <SCThemeProvider theme={{ yoga: getTheme(theme) }}>
    {children}
  </SCThemeProvider>
);

ThemeProvider.propTypes = {
  theme: PropTypes.string,
  locale: PropTypes.string,
  children: PropTypes.node.isRequired,
};

ThemeProvider.defaultProps = {
  theme: 'endUser',
  locale: 'pt-BR',
};

export default ThemeProvider;
