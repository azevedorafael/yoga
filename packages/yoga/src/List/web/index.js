import ListLinkItem from './ListLinkItem';
import ListItem from './ListItem';
import List from './List';

export { List, ListItem, ListLinkItem };
