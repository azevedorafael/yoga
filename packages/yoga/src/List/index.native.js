import { List, ListItem } from './native';

List.Item = ListItem;

export default List;
