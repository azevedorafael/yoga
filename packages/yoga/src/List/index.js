import { List, ListItem } from './web';

List.Item = ListItem;

export default List;
