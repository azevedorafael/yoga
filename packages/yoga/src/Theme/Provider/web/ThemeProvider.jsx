import React from 'react';
import { node } from 'prop-types';

import Provider from '../Provider';
import GlobalStyle from './GlobalStyle';

/** This component provides a theme to all React components underneath itself via the context API. */
const ThemeProvider = ({ children, ...rest }) => (
  <Provider {...rest}>
    <GlobalStyle />
    {children}
  </Provider>
);

ThemeProvider.propTypes = {
  children: node.isRequired,
};

export default ThemeProvider;
