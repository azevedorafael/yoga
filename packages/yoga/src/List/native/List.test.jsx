import React from 'react';
import { render } from '@testing-library/react-native';
import { Text } from 'react-native';
import ThemeProvider from '../../ThemeProvider';
import List from '..';

const data = [
  { key: 'Devin' },
  { key: 'Dan' },
  { key: 'Dominic' },
  { key: 'Jackson' },
  { key: 'James' },
  { key: 'Joel' },
  { key: 'John' },
  { key: 'Jillian' },
  { key: 'Jimmy' },
  { key: 'Julie' },
];

describe('<List />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with a default list', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with a horizontal list', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              horizontal
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
