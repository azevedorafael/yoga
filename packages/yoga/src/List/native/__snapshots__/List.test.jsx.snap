// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Snapshots Without props should match snapshot with a default list 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    data={
      Array [
        Object {
          "key": "Devin",
        },
        Object {
          "key": "Dan",
        },
        Object {
          "key": "Dominic",
        },
        Object {
          "key": "Jackson",
        },
        Object {
          "key": "James",
        },
        Object {
          "key": "Joel",
        },
        Object {
          "key": "John",
        },
        Object {
          "key": "Jillian",
        },
        Object {
          "key": "Jimmy",
        },
        Object {
          "key": "Julie",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      style={null}
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;

exports[`<List /> Snapshots Without props should match snapshot with a horizontal list 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    data={
      Array [
        Object {
          "key": "Devin",
        },
        Object {
          "key": "Dan",
        },
        Object {
          "key": "Dominic",
        },
        Object {
          "key": "Jackson",
        },
        Object {
          "key": "James",
        },
        Object {
          "key": "Joel",
        },
        Object {
          "key": "John",
        },
        Object {
          "key": "Jillian",
        },
        Object {
          "key": "Jimmy",
        },
        Object {
          "key": "Julie",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#DBDBE8",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
